### PR TITLE
add contribution admin in frontend 

### PR DESCRIPTION
**Pull request by wenjzhu**
## Description
add admin names in the frontend form submission when contributors are creating a new contribution.  

**Fixes #(add issue number here and remove parentheses)**

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Immediately
- [ ] Within a week
- [x] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [ ] I have updated the [CHANGELOG](../CHANGELOG.md).
- [ ] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

### CODE REVIEW

**Comment by sandeep-ps** [comment]
@wenjzhu, please fix the merge conflicts in this PR. There seem to be file changes not related to this PR. Hopefully, it should go away after fixing merge conflicts. Thanks.

File: contributions/catalog/controllers/contribute.py [hunk 1]
```diff
@@ -62,8 +62,8 @@ def create():
 
         # add contributionAdmins to the json_contiubtion
         contribution = jsonutil.add_contribution_admins(contribution)
-
         json_contribution = json.dumps(contribution, indent=4)
+        print(json_contribution)
```
Comment (sandeep-ps) [inline contributions/catalog/controllers/contribute.py:66]: Please remove print statements. You may use logging instead. Thanks.

File: lib/auth-middleware/auth_middleware/__init__.py [hunk 2]
```diff
@@ -23,6 +23,7 @@
 import requests
 from connexion.exceptions import OAuthProblem
 from flask import request, g
+from requests_oauthlib import OAuth2Session
```
Comment (sandeep-ps) [inline lib/auth-middleware/auth_middleware/__init__.py:26]: Changes to lib/ folder shouldn't be part of this PR, right? It may have got added when resolving merge conflicts. Could you please take a look?

File: lib/auth-middleware/auth_middleware/__init__.py [hunk 3]
```diff
@@ -46,6 +47,7 @@
 uiucedu_is_member_of = "uiucedu_is_member_of"
 DEBUG_ON = False
 
+GITHUB_CLIENT_ID = os.getenv('GITHUB_CLIENT_ID')
```
Comment (sandeep-ps) [inline lib/auth-middleware/auth_middleware/__init__.py:50]: Same here: Changes to lib/ folder shouldn't be part of this PR, right?

File: lib/auth-middleware/setup.py [hunk 4]
```diff
@@ -27,9 +27,11 @@
     description='Rokwire Platform Auth Middleware',
     # long_description=open('README.txt').read(),
     install_requires=[
-        'PyJWT>=1.7.1,<2',
-        'requests>=2.22.0,<3',
-        'cryptography>=3,<4',
+        'PyJWT>=1.7.1',
+        'requests>=2.22.0',
+        'cryptography>=2.7',
+        'requests_oauthlib>=1.3.0'
```
Comment (ywkim312) [inline lib/auth-middleware/setup.py:33]: This one can be gone. Probably, we don't need any auth-middleware change at all in here.

File: lib/auth-middleware/auth_middleware/__init__.py [hunk 5]
```diff
@@ -309,3 +311,28 @@ def use_security_token_auth(func):
 #         abort(401)
 #     request.user_token_data = id_info
 #     return
+
+def verify_githubauth(token_str):
```
Comment (ywkim312) [inline lib/auth-middleware/auth_middleware/__init__.py:315]: Whole block will not be used. Like @sandeep-ps mentioned, lib/auth-middleware don't need be changed at all. Sorry for the confusion.

File: contributions/catalog/controllers/contribute.py (hunk 1 above)
Comment (wenjzhu) [reply contributions/catalog/controllers/contribute.py]: removed.

File: lib/auth-middleware/auth_middleware/__init__.py (hunk 3 above)
Comment (wenjzhu) [reply lib/auth-middleware/auth_middleware/__init__.py:50]: Should we remove all the changes in auth-middleware then?

File: lib/auth-middleware/setup.py (hunk 4 above)
Comment (wenjzhu) [reply lib/auth-middleware/setup.py:33]: ok, I see, then i can probably remove them

File: lib/auth-middleware/auth_middleware/__init__.py (hunk 3 above)
Comment (ywkim312) [reply lib/auth-middleware/auth_middleware/__init__.py:50]: Yes, you probably want to revert to the previous one.

**Comment by sandeep-ps** [comment]
@wenjzhu, the basic functionality (being able to add GitHub usernames as contribution admins) looks fine to me. As follow-up tasks, we need to improve this basic functionality by filling up the logged-in user's GitHub username by default, maybe use autocomplete feature using GitHub API for searching users (https://docs.github.com/en/rest/reference/search#search-users). Let's discuss this during the next technical meeting.

@ywkim312, any other comments?

**Review by sandeep-ps — APPROVED** [review]
